source/extensions/filters/http/jwt_authn/jwks_async_fetcher.cc

```diff
@@ -64,6 +96,11 @@ void JwksAsyncFetcher::handleFetchDone() {
     init_target_.reset();
   }
 
+  if (backoff_strategy_) {
+    backoff_strategy_->reset();
```

Comment:
With this,  retry policy only works for the first fetch,   it will not work for the re-fetch after cache duration.

Reply:
this is not a reset of [the unique pointer](https://github.com/alichnewsky/envoy/blob/8f18c66ffdc778e23cd2c61a3647fdb3405f32e2/envoy/common/backoff_strategy.h#L27), it is a call to the unfortunately named [backoff strategy reset() member function](https://github.com/alichnewsky/envoy/blob/8f18c66ffdc778e23cd2c61a3647fdb3405f32e2/envoy/common/backoff_strategy.h#L24)

after the first `fetch()` success ( on cache duration timer or retry timer ), or on failure after having already executed the maximum number of retries, `handlFetchDone()` is called, and the [backoff strategy is reset to its first interval ](https://github.com/alichnewsky/envoy/blob/8f18c66ffdc778e23cd2c61a3647fdb3405f32e2/source/common/common/backoff_strategy.cc#L25) and we retry with a small interval at next cache duration?

and with all attempts available ( next line)